src/net/sourceforge/kolmafia/session/BastilleBattalionManager.java

```diff
@@ -0,0 +1,922 @@
+  }
+
+  private static void clearChoices() {
+    Preferences.setString("_bastilleChoice1", "");
+    Preferences.setString("_bastilleChoice2", "");
+    Preferences.setString("_bastilleChoice3", "");
+  }
+
+  private static void getChoices(String responseText) {
+    Map<Integer, String> choices = ChoiceUtilities.parseChoices(responseText);
+    if (choices.size() == 3) {
+      Preferences.setString("_bastilleChoice1", choices.get(1));
+      Preferences.setString("_bastilleChoice2", choices.get(2));
+      Preferences.setString("_bastilleChoice3", choices.get(3));
+    }
+  }
+
+  private static void endGame(String text) {
+    Preferences.increment("_bastilleGames");
+    Preferences.setInteger("_bastilleGameTurn", 0);
+  }
+
+  // *** Logging
+
+  private static void logLine(String message) {
+    RequestLogger.printLine(message);
+    RequestLogger.updateSessionLog(message);
+  }
+
+  private static AdventureResult SHARK_TOOTH_GRIN = EffectPool.get(EffectPool.SHARK_TOOTH_GRIN);
+  private static AdventureResult BOILING_DETERMINATION =
+      EffectPool.get(EffectPool.BOILING_DETERMINATION);
+  private static AdventureResult ENHANCED_INTERROGATION =
+      EffectPool.get(EffectPool.ENHANCED_INTERROGATION);
+
+  public static void logBoosts() {
+    StringBuilder buf = new StringBuilder();
+    if (KoLConstants.activeEffects.contains(SHARK_TOOTH_GRIN)) {
+      logLine("(Military attack and defense boosted from Shark Tooth Grin)");
+      buf.append('M');
+    }
+    if (KoLConstants.activeEffects.contains(BOILING_DETERMINATION)) {
+      logLine("(Castle attack and defense boosted from Boiling Determination)");
+      buf.append('C');
+    }
+    if (KoLConstants.activeEffects.contains(ENHANCED_INTERROGATION)) {
+      logLine("(Psychological attack and defense boosted from Enhanced Interrogation)");
+      buf.append('P');
+    }
+    Preferences.setString("_bastilleBoosts", buf.toString());
+  }
+
+  private static void logStrength() {
+    StringBuilder buf = new StringBuilder();
+    buf.append("Military ");
+    buf.append(currentStats.get(Stat.MA));
+    buf.append("/");
+    buf.append(currentStats.get(Stat.MD));
+    buf.append(" ");
+    buf.append("Castle ");
+    buf.append(currentStats.get(Stat.CA));
+    buf.append("/");
+    buf.append(currentStats.get(Stat.CD));
+    buf.append(" ");
+    buf.append("Psychological ");
+    buf.append(currentStats.get(Stat.PA));
+    buf.append("/");
+    buf.append(currentStats.get(Stat.PD));
+    String message = buf.toString();
+    RequestLogger.printLine(message);
+    RequestLogger.updateSessionLog(message);
+  }
+
+  private static StringBuilder logAction(StringBuilder buf, String action) {
+    buf.append("Turn #");
+    buf.append(Preferences.getInteger("_bastilleGameTurn"));
+    buf.append(": ");
+    buf.append(action);
+    return buf;
+  }
+
+  // *** Interface for testing
+
+  public static Map<Upgrade, Style> getCurrentStyles() {
+    return currentStyles;
```

Comment:
doesn't appear to be called anywhere?

Reply:
Good point. I thought I would use it in tests, but when I got around to writing them, I used the "black box" approach - I only looked at what the module exposed through its user interface - the set of properties that reflect internal state.

I will remove this unused method and add a new property - _bastilleCurrentStyles - and will modify my "configuration" test to use that in validation.

Reply:
I did that. Here are updated comments from the reset() method:
```
    // When you initially visit the control rig, you can select the "style" of
    // the four available upgrades: barbican, drawbridge, murder holes, moat.
    // You can fiddle with them to your heart's content until you start your
    // first game of the day. At that point they are locked in and you will
    // receive the appropriate prizes at the end of that game, win or lose.
    Preferences.setString("_bastilleCurrentStyles", "");

    // Each configured style grants a specific bonus to the set of stats.  That
    // is locked in once you start your first game. As you progress through the
    // game, you perform actions to add or subtract to specific (or all) attack
    // or defense stats. Stats revert to your style-provided bonuses at the
    // start of each game.
    //
    // Since we don't know your actual stats; this is user visible bonuses.
    Preferences.setString("_bastilleStats", "");
```
This is in now and the tests use it, rather than internal methods.